fedimint-client/src/lib.rs

```diff
@@ -792,19 +776,14 @@ impl ClientWeak {
 /// `Arc<Client>`s such that at least one `Executor` never gets dropped.
 impl Drop for ClientHandle {
     fn drop(&mut self) {
-        if self.inner.is_none() {
```

Comment:
Note for reviewer: Previously, `self.inner` being `None` indicated that the executor was already stopped. `Since `inner` is no longer an `Option` and we don't ever return early here anymore, the call to `stop_executor()` below can be called after the executor has already been stopped. This will result in the following code being hit:

https://github.com/fedimint/fedimint/blob/7b7bf0f45574911e8348c68f2abc5c61b36b117c/fedimint-client/src/sm/executor.rs#L126-L130

This will produce an extra log statement, but doesn't affect anything otherwise.